File: import_ir.py
from span import SourceSpan
from typing import Union
from expression_ir import IdentifiedIRNode


class ImportIR(IdentifiedIRNode):
    def __init__(self, id: int, local_symbol_id: int, scope_id: int, kind: str, module_name: str, imported_name: Union[int, None], alias: Union[str, None], relative_level: int, span: Union[SourceSpan, None]):
        super().__init__(id=id, span=span)
        self.id=id
        self.local_symbol_id=local_symbol_id
        self.scope_id=scope_id
        self.kind=kind
        self.module_name=module_name
        self.imported_name=imported_name
        self.alias=alias
        self.relative_level=relative_level
        self.span=span

File: rust/src/ir/nodes/import_ir.rs
use crate::ir::nodes::SourceSpan;

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum ImportKind {
    Import,
    FromImport,
    Unknown(i32),
}

impl From<i32> for ImportKind {
    fn from(value: i32) -> Self {
        match value {
            1 => Self::Import,
            2 => Self::FromImport,
            other => Self::Unknown(other),
        }
    }
}

#[derive(Debug, Clone)]
pub struct ImportIR {
    pub id: i64,
    pub local_symbol_id: i64,
    pub scope_id: i64,
    pub kind: ImportKind,
    pub module_name: String,
    pub imported_name: Option<String>,
    pub alias: Option<String>,
    pub relative_level: i64,
    pub span: Option<SourceSpan>,
}
